http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/response/CrtResponseAdapter.java

```diff
@@ -129,9 +129,16 @@ private void onSuccessfulResponseComplete(HttpStream stream) {
     private void onFailedResponseComplete(HttpStream stream, HttpException error) {
         log.debug(() -> "HTTP response encountered an error.", error);
 
-        IOException wrappedError = new IOException(error);
-        responsePublisher.error(wrappedError);
-        failResponseHandlerAndFuture(stream, wrappedError);
+        Throwable toThrow = error;
+
+        if (HttpClientConnection.isErrorRetryable(error)) {
+            // IOExceptions get retried, and if the CRT says this error is retryable,
+            // it's semantically an IOException anyway.
+            toThrow = new IOException(error);
+        }
+
+        responsePublisher.error(toThrow);
+        failResponseHandlerAndFuture(stream, toThrow);
     }
```

Comment:
Do we need to apply the same logic here? https://github.com/aws/aws-sdk-java-v2/blob/master/http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/CrtRequestExecutor.java#L75

Reply:
I think this should be fine. Whatever exception gets thrown in connection startup time is going to be an IOException.

Line 93 of that same file, however, needs an update. Good catch!